data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/reconciler/impl/ResourcesReconcilerImpl.java

```diff
@@ -60,8 +60,8 @@ public class ResourcesReconcilerImpl implements ResourcesReconciler {
     this.ingressReconcilerListener = ingressReconcilerListener;
     this.egressReconcilerListener = egressReconcilerListener;
 
-    this.cachedResources = new ConcurrentHashMap<>(CACHED_RESOURCES_INITIAL_CAPACITY);
-    this.cachedEgresses = new ConcurrentHashMap<>(egressReconcilerListener != null ? CACHED_RESOURCES_INITIAL_CAPACITY : 0);
+    this.cachedResources = ingressReconcilerListener == null ? null : new ConcurrentHashMap<>(CACHED_RESOURCES_INITIAL_CAPACITY);
```

Comment:
aren't we doing a NPE check a little above?

Reply:
No, the condition is `if (ingressReconcilerListener == null && egressReconcilerListener == null) {`

Reply:
so this is handling the case of one of them null

Reply:
:+1: